### PR TITLE
Capture configuration files & runtime config in support bundle

### PR DESCRIPTION
**Pull request by dehaansa**
#### PR Description
Add configuration sources (local file & remote config) to the support bundle.

One thing that requires consideration -- this is not currently doing any redaction of the configuration files, should we make this opt-in/opt-out rather than always collected?

#### Which issue(s) this PR fixes
https://github.com/grafana/alloy/issues/2066

#### PR Checklist

- [X] CHANGELOG.md updated
- [X] Documentation added

### CODE REVIEW

**Comment by mattdurham** [comment]
Ideally we should replace secret values with a placeholder.

**Comment by mattdurham** [comment]
In the old agent one we used a custom yaml marshaller to redact the secrets, I feel we need to also redact the values here.

File: docs/sources/troubleshoot/support_bundle.md [hunk 1]
```diff
@@ -44,8 +44,11 @@ A support bundle contains the following data:
 * `alloy-runtime-flags.txt` contains the values of the runtime flags available in {{< param "PRODUCT_NAME" >}}.
 * The `pprof/` directory contains Go runtime profiling data (CPU, heap, goroutine, mutex, block profiles) as exported by the pprof package.
 Refer to the [profile][profile] documentation for more details on how to use this information.
+* The `sources/` directory contains copies of the local configuration files being used to configure {{< param "PRODUCT_NAME" >}}.
```
Comment (clayton-cornell) [inline docs/sources/troubleshoot/support_bundle.md:47]:
```suggestion
* The `sources/` directory contains copies of the local configuration files used to configure {{< param "PRODUCT_NAME" >}}.
```

**Comment by dehaansa** [comment]
Implemented a POC of redacting secrets in the AST & printer, but I need to consider how it affects the fmt command and other uses of printer.

File: docs/sources/troubleshoot/support_bundle.md [hunk 2]
```diff
@@ -45,8 +45,11 @@ A support bundle contains the following data:
 * `alloy-runtime-flags.txt` contains the values of the runtime flags available in {{< param "PRODUCT_NAME" >}}.
 * The `pprof/` directory contains Go runtime profiling data (CPU, heap, goroutine, mutex, block profiles) as exported by the pprof package.
 Refer to the [profile][profile] documentation for more details on how to use this information.
+* The `sources/` directory contains copies of the local configuration files used to configure {{< param "PRODUCT_NAME" >}}.
```
Comment (mattdurham) [inline docs/sources/troubleshoot/support_bundle.md:48]: Does this include local modules? Totally fine if it doesnt but may want to add a bit more information.

**Review by mattdurham — APPROVED** [review]
LGTM, minor documentation question

**Pull request merged**